openapi3filter/req_resp_decoder.go

```diff
@@ -595,3 +596,47 @@ func parsePrimitive(raw string, schema *openapi3.SchemaRef) (interface{}, error)
 		panic(fmt.Sprintf("schema has non primitive type %q", schema.Value.Type))
 	}
 }
+
+// BodyDecoder is an interface to decode a body of a request or response.
+// An implementation must return a value that is a primitive, []interface{}, or map[string]interface{}.
+type BodyDecoder func(data []byte) (interface{}, error)
+
+// bodyDecoders contains decoders for supported content types of a body.
+// By default, there is content type "application/json" is supported only.
+var bodyDecoders = map[string]BodyDecoder{
+	"plain/text": func(body []byte) (interface{}, error) {
+		return string(body), nil
+	},
+	"application/json": func(body []byte) (interface{}, error) {
+		var value interface{}
+		if err := json.Unmarshal(body, &value); err != nil {
+			return nil, err
+		}
+		return value, nil
+	},
+}
+
+// RegisterBodyDecoder registers a request body's decoder for a content types.
+//
+// If a decoder for the specified content type is already exists, the function replaces
+// it with the specified decoder.
+func RegisterBodyDecoder(contentType string, decoder BodyDecoder) {
```

Comment:
Nit: we should also have
```golang
// UnregisterBodyDecoder dissociates a body decoder from a content type.
//
// Decoding this content type will result in an error.
func UnregisterBodyDecoder(contentType string) {
	delete(bodyDecoders, contentType)
}
```
Same thing here: if you feel like this should complain on error, add some checks ;)

Reply:
Done.